src/utils/nodeUtils.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+
+export interface NodePackage {
+    main?: string;
+    name?: string;
+    scripts?: { [key: string]: string };
+}
+
+export type InspectMode = 'default' | 'break';
+
+export async function readPackage(packagePath: string): Promise<NodePackage> {
+    return <NodePackage> await fse.readJson(packagePath);
+}
+
+export async function inferPackageName(nodePackage: NodePackage | undefined, packagePath: string): Promise<string> {
+    if (nodePackage && nodePackage.name !== undefined) {
```

Comment:
nit: If there's an empty but not-undefined name this will return it. Is that expected? As an alternative,

return (nodePackage && nodePackage.name) || path.basename(path.dirname(packagePath));

Reply:
Yeah, not sure why I insisted on `undefined` there.  Updated to account for an empty string.